### PR TITLE
Removed backticks from mermaid code

### PR DESCRIPTION
**Pull request by subhajit20**
# Closes #<CORRESPONDING ISSUE NUMBER>

### Changes
Removed backticks from mermaid code
Fix - https://github.com/accordproject/concerto/issues/665
### Flags

- <ONE>
- <TWO>

### Related Issues
- Issue #<NUMBER>
- Pull Request #<NUMBER>

### Author Checklist
- [ ] Ensure you provide a [DCO sign-off](https://github.com/probot/dco#how-it-works) for your commits using the `--signoff` option of git commit.
- [ ] Vital features and changes captured in unit and/or integration tests
- [ ] Commits messages follow [AP format](https://github.com/accordproject/techdocs/blob/master/DEVELOPERS.md#commit-message-format)
- [ ] Extend the documentation, if necessary
- [ ] Merging to `main` from `fork:branchname`

### CODE REVIEW

**Comment by subhajit20** [comment]
Hey @sanketshevkar, is this solution meaningful or are there any changes needed?

**Comment by ekarademir** [comment]
@subhajit20 Could you sign your PRs please? https://github.com/dcoapp/app#how-it-works

**Comment by subhajit20** [comment]
Signed-off-by: Subhajit Ghosh subhajitstd07@gmail.comn

**Comment by subhajit20** [comment]
> @subhajit20 Could you sign your PRs please? https://github.com/dcoapp/app#how-it-works

Hey Done that.

File: lib/codegen/fromcto/mermaid/mermaidvisitor.js
Comment (subhajit20) [reply lib/codegen/fromcto/mermaid/mermaidvisitor.js:52]: Cool, I will do that
Thanks!!

Comment (subhajit20) [reply lib/codegen/fromcto/mermaid/mermaidvisitor.js]: Hey @sanketshevkar I have made certain changes and updated the snapshots.
Can you check that once and let me know if further changes are needed?

Comment (sanketshevkar) [reply lib/codegen/fromcto/mermaid/mermaidvisitor.js]: @subhajit20 I don't think so this solves the issue, earlier you had removed back-ticks for classDeclaration name for declarations which has properties.
Now you have you removed back-ticks for classDeclaration name for declarations without properties.

Please test your changes on the editor I have shared.

Comment (subhajit20) [reply lib/codegen/fromcto/mermaid/mermaidvisitor.js]: Hey @sanketshevkar can you please check once I have made certain changes.

**Review by sanketshevkar — CHANGES_REQUESTED** [review]

**Comment by subhajit20** [comment]
hey @sanketshevkar I think this commit can removes backticks from properties

**Comment by sanketshevkar** [comment]
Can you please share some sample model and its output mermaid diagram on this [editor ](https://mermaid.live/edit#pako:eNqFkE1rwzAMhv-K0XWmNG1pOlPKoLsOBr0NHyxstTPEcnAcWBby3-dkuY6hg74fXmkEGx2BAttg1716fCQMmpdMmEApoHcv1WZb7J1SF9mIUfP5LGxkS22-XDQLIZ6EueXk-WGE8c5onjT_gbnG0CIP_3MYA80kkLASis5xHtOQPymQBlVCR3fsm6zhdxT7HG8DW1A59SShbx1mWi8DdcemK9WiANQIX6C2EgZQ1X5zOB6r59Oh3p3qaldPEr5jLAuVBHI-x_S2Pmp2C-Bj6S_A6Qcb5WwN)

Also @mttrbrts should we add a test case to validate this?

**Comment by sanketshevkar** [comment]
PR LGTM!

@mttrbrts @dselman can you please review the PR once?

**Comment by mttrbrts** [comment]
Thanks for your contribution, @subhajit20 . The output is now looking much better! 🤩 

When I render one of the test cases in a mermaid editor I see that the properties are rendered without backticks, but that enum declarations values still have them. Can you make a similar change for that node too, please?

image

@sanketshevkar

**Comment by subhajit20** [comment]
hey @mttrbrts done that.
Thanks!!f

**Comment by sanketshevkar** [comment]
Good job @subhajit20 !! :)
LGTM!

**Comment by sanketshevkar** [comment]
Let's wait for @mttrbrts's approval.

**Comment by dselman** [comment]
Nice work @subhajit20. When are back ticks required? I.e. what characters does Mermaid require to be escaped with back ticks?

Can you please test this with our "kitchen sink" model and check the rendered output?
https://github.com/accordproject/concerto-codegen/blob/main/test/codegen/fromcto/data/model/hr.cto

**Comment by subhajit20** [comment]
hey @sanketshevkar @dselman is there anything further I have to do in this pr?

**Comment by subhajit20** [comment]
I just run the mentioned by @dselman  test using `npx mocha` and log the output  and all the properties of classes and enums are rendered without backticks here are two screenshots
- enum
Screenshot 2024-03-15 at 9 05 09 PM

- class
Screenshot 2024-03-15 at 9 04 12 PM

**Comment by subhajit20** [comment]
And here is the mermaid editor output
Screenshot 2024-03-15 at 9 11 52 PM

**Review by sanketshevkar — APPROVED** [review]

**Review by mttrbrts — APPROVED** [review]

**Pull request merged**
